File: src/entities/Door.ts
import {
  Collection,
  Entity,
  OneToMany,
  PrimaryKey,
  Property,
  QueryOrder,
} from '@mikro-orm/core';
import { Sequence } from './Sequence';

@Entity()
export class Door {
  @PrimaryKey({ autoincrement: false })
  id: number;

  @Property()
  label: string;

  @Property({ default: true })
  isEnabled: boolean;

  @Property()
  state: 'open' | 'closed';

  @OneToMany(() => Sequence, (sequence) => sequence.door, {
    orderBy: { index: QueryOrder.ASC },
  })
  sequences = new Collection<Sequence>(this);
}

File: app/lib/models/door.dart
class Door {
  final int id;
  final String label;
  final bool isEnabled;
  final String state;
  final int openDuration;
  final int closeDuration;

  const Door({
    required this.id,
    required this.label,
    required this.isEnabled,
    required this.state,
    required this.openDuration,
    required this.closeDuration,
  });

  Door.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        label = json['label'],
        isEnabled = json['isEnabled'],
        state = json['state'] as String,
        openDuration = json['openDuration'],
        closeDuration = json['closeDuration'];
}
